src/MUSI6106Exec/tests.h

```diff
@@ -0,0 +1,180 @@
+    std::string filterName;
+    if (filterType == CCombFilterIf::kCombIIR)
+    {
+        filterName = "IIR";
+    }
+    else if (filterType == CCombFilterIf::kCombFIR)
+    {
+        filterName = "FIR";
+    }
+    std::string sOutputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/output/6_"+filterName+"filtered.wav";
+    float gain = 0.5;
+    float delay = 0.01;
+//    CCombFilterIf::CombFilterType_t filterType = CCombFilterIf::kCombIIR;
+    Error_t test3error = run_filtering(sInputFilePath,sOutputFilePath,filterType,gain,delay,blockSize);
+    cout << "Test with blocksize: " << blockSize << " and filter type: " << filterName << " successful!" << endl;
+//    cout << "Test with blocksize: " << blockSize << " successful!" << endl;
+    assert (test3error == Error_t::kNoError);
+    return test3error;
+}
+
+Error_t test4(CCombFilterIf::CombFilterType_t filterType)
+{
+//    std::string sInputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/silence.wav";
+//    std::string sOutputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/output/silence_filtered.wav";
+//    std::string filterName;
+//    if (filterType == CCombFilterIf::kCombIIR)
+//    {
+//        filterName = "IIR";
+//    }
+//    else if (filterType == CCombFilterIf::kCombFIR)
+//    {
+//        filterName = "FIR";
+//    }
+//    float gain = 0.5;
+//    float delay = 0.01;
+//    int blockSize = 1024;
+//
+//    Error_t test4error = run_filtering(sInputFilePath,sOutputFilePath,filterType,gain,delay,blockSize);
+    int iNumChannels = 2;
+    int kBlockSize = 1024;
+    int iNumFrames = kBlockSize;
+    float **ppfInputBuffer = nullptr;
+    float **ppfOutputBuffer = nullptr;
+
+    CCombFilterIf *filter = nullptr;
+    CCombFilterIf::create(filter);
+    filter->init(filterType,0.01,44100,iNumChannels, 0.5);
+    ppfInputBuffer = new float*[iNumChannels];
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        ppfInputBuffer[i] = new float[kBlockSize];
+    }
+    ppfOutputBuffer = new float*[iNumChannels];
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        ppfOutputBuffer[i] = new float[kBlockSize];
+    }
+
+    //Set input buffer to zeros
+    for (int c = 0; c < iNumChannels; c++)
+    {
+        for(int i = 0; i < iNumFrames;i++)
+        {
+            ppfInputBuffer[c][i] = 0;
+        }
+    }
+    //Process
+    Error_t test4 = filter->process(ppfInputBuffer,ppfOutputBuffer,iNumFrames);
+    //Clear memory
+    CCombFilterIf::destroy(filter);
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        delete[] ppfInputBuffer[i];
+        delete[] ppfOutputBuffer[i];
+    }
+    delete[] ppfInputBuffer;
+    delete[] ppfOutputBuffer;
+    ppfInputBuffer = nullptr;
+    ppfOutputBuffer = nullptr;
+    assert (test4 == Error_t::kNoError);
+    cout << "Test with silence successful!" << endl;
```

Comment:
These tests could be more useful if they checked the output of the effect for correctness (e.g. in this case, make sure that the output is also silent), rather than just making sure that the function calls don't return errors.